### ISSUE TITLE
Clarify violation in `filename` rule when class contains a single class and (extension) functions

### ISSUE THREAD

**Issue by FloEdelmann** [pinterest/ktlint#2998]
## Expected Behavior

Given this file: https://github.com/streetcomplete/StreetComplete/blob/5dc586dc5bb1ca72e44d02232de78a14d571e09a/app/src/commonMain/kotlin/de/westnordost/streetcomplete/osm/WaysCrossingUtils.kt

It is called `WaysCrossingUtils.kt`, contains a single `data class WaysCrossing(…)`, a public `fun findNodesAtCrossingsOf(…)` and some private functions and extension functions. Since it contains no `WaysCrossing` extension function, I expected the [`filename` rule](https://pinterest.github.io/ktlint/latest/rules/standard/#file-name) not to trigger.

To quote https://github.com/streetcomplete/StreetComplete/commit/4ed305009c42c3617a17bf06c57fa2a40e60d353#commitcomment-158225965:

> The `WaysCrossing` is just the result data type of this utility function. Using the `Utils` postfix is more consistent here for this.
> 
> It could be named `WaysCrossing` if the API was like this:
> 
> ```kotlin
> data class WaysCrossing(...) {
>    // ...
> 
>   companion object {
>     fun find(barrierWays: Sequence<Way>, movingWays: Sequence<Way>, mapData: MapData) : WaysCrossing
>   }
> }
> ```
> 
> but it isn't, and I think it makes sense that it isn't.

## Current Behavior

A lint violation is reported for the [`filename` rule](https://pinterest.github.io/ktlint/latest/rules/standard/#file-name):

```
…/WaysCrossingUtils.kt:1:1: File 'WaysCrossingUtils.kt' contains a single class and possibly also extension functions for that class and should be named same after that class 'WaysCrossing.kt' (cannot be auto-corrected) (standard:filename)
```

## Additional information

* Current version of ktlint: 1.6.0
